ql/src/java/org/apache/hadoop/hive/ql/parse/StorageFormat.java

```diff
@@ -81,11 +84,34 @@ public boolean fillStorageFormat(ASTNode child) throws SemanticException {
       }
       break;
     case HiveParser.TOK_STORAGEHANDLER:
-      storageHandler = processStorageHandler(child.getChild(0).getText());
-      if (child.getChildCount() == 2) {
-        BaseSemanticAnalyzer.readProps(
-          (ASTNode) (child.getChild(1).getChild(0)),
-          serdeProps);
+      for (int i = 0; i < child.getChildCount(); i++) {
```

Comment:
What happens when StorageHandler is HBase and Stored By is also set?

Reply:
In this case, the `storagehandler.getFileFormatPropertyKey` would return `null` and the fileformat property would be disregarded. But I see your point here, maybe we should warn the end user about this.